File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        
        for(int i=0; i<s.length(); i++) {
            int cint = s.charAt(i);
            for(int j=1; j<=index;) {
                if(++cint>'z') {
                    cint -= 26;
                }
                if(skip.contains(String.valueOf((char)cint))) {
                    continue;
                }
                j++;
            }
            answer += (char)cint;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    const offsets = Array.from({length: 123}, () => 0);
    
    for (const char of skip.split("")) {
        offsets[char.charCodeAt()] = 1;
    }
    
    const result = [];
    
    for (const char of s.split("")) {
        let cnt = index;
        let ascii = char.charCodeAt();
        while (cnt-- > 0) {
            if (++ascii === 123) {
                ascii = 97;
            }
            cnt += offsets[ascii];
        }
        result.push(String.fromCharCode(ascii));
    }
    
    return result.join("");
}